File: Documentation/5.2/Samples/nodejs/ClientApi/Session/Querying/TextSearch/exactMatchSearch.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();
const session = documentStore.openSession();

{
    let fieldName, value, exact;
    const query = session.query();
    //region query_1_0
    query.whereEquals(fieldName, value, [exact]);
    query.whereNotEquals(fieldName, value, [exact]);

    // ... rest of where methods
    //endregion
}

async function examples() {
    {
        //region query_1_1
        // load all entities from 'Employees' collection
        // where firstName equals 'Robert' (case sensitive match)
        const employees = await session.query({ collection: "Employees" })
            .whereEquals("FirstName", "Robert", true)
            .all();
        //endregion
    }

    {
        //region query_2_1
        // return all entities from 'Orders' collection
        // which contain at least one order line with
        // 'Singaporean Hokkien Fried Mee' product
        // perform a case-sensitive match
        const orders = await session.query({ collection: "Orders" })
            .whereEquals("Lines[].ProductName", "Singaporean Hokkien Fried Mee", true)
            .all();
        //endregion
    }
}

File: Documentation/5.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/Querying/TextSearch/ExactMatchSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Linq;
using Raven.Documentation.Samples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session.Querying.TextSearch
{
    public class ExactMatchSearch
    {
        private interface IFoo<T>
        {
            #region query_1_0
            IRavenQueryable<T> Where<T>(Expression<Func<T, int, bool>> predicate, bool exact);

            IRavenQueryable<T> Where<T>(Expression<Func<T, bool>> predicate, bool exact);
            #endregion
        }

        public async Task Examples()
        {
            using (var store = new DocumentStore())
            {
                using (var session = store.OpenSession())
                {
                    #region query_1_1
                    // load all entities from 'Employees' collection
                    // where FirstName equals 'Robert' (case sensitive match)
                    List<Employee> employees = session
                        .Query<Employee>()
                        .Where(x => x.FirstName == "Robert", exact: true)
                        .ToList();
                    #endregion
                }

                using (var asyncSession = store.OpenAsyncSession())
                {
                    #region query_1_1_async
                    // load all entities from 'Employees' collection
                    // where FirstName equals 'Robert' (case sensitive match)
                    List<Employee> employees = await asyncSession
                        .Query<Employee>()
                        .Where(x => x.FirstName == "Robert", exact: true)
                        .ToListAsync();
                    #endregion
                }

                using (var session = store.OpenSession())
                {
                    #region query_2_1
                    // return all entities from 'Orders' collection
                    // which contain at least one order line with
                    // 'Singaporean Hokkien Fried Mee' product
                    // perform a case-sensitive match
                    List<Order> orders = session
                        .Query<Order>()
                        .Where(x => x.Lines.Any(p => p.ProductName == "Singaporean Hokkien Fried Mee"), exact: true)
                        .ToList();
                    #endregion
                }

                using (var asyncSession = store.OpenAsyncSession())
                {
                    #region query_2_1_async
                    // return all entities from 'Orders' collection
                    // which contain at least one order line with
                    // 'Singaporean Hokkien Fried Mee' product
                    // perform a case-sensitive match
                    List<Order> orders = await asyncSession
                        .Query<Order>()
                        .Where(x => x.Lines.Any(p => p.ProductName == "Singaporean Hokkien Fried Mee"), exact: true)
                        .ToListAsync();
                    #endregion
                }
            }
        }
    }
}
